src/main/java/de/craften/ui/swingmaterial/Utils.java

```diff
@@ -0,0 +1,185 @@
+        Method getMethod = null;
+        try {
+            Class sunGE = Class.forName("sun.java2d.SunGraphicsEnvironment");
+            Method[] meths = sunGE.getDeclaredMethods();
+            for (Method meth:meths) {
+                if (meth.getName().equals("getUsableBounds")
+                    && Arrays.equals(meth.getParameterTypes(),
+                            new Class[]{java.awt.GraphicsDevice.class})
+                    && meth.getExceptionTypes().length == 0
+                    && meth.getReturnType()
+                            .equals(java.awt.Rectangle.class)) {
+                    //We found it!
+                    getMethod = meth;
+                    found = true;
+                    break;
+                }
+            }
+        } catch (ClassNotFoundException ex) {
+            //It seems not
+            found = false;
+        }
+        useSun2D = found;
+        getUsableBounds = getMethod;
+    }
+    
+    /**
+     * Checks the area available in the desktop, excluding the taskbar.
+     * In order to do this, an attempt to call
+     * <code>sun.java2d.SunGraphicsEnvironment.getUsableBounds()</code> is
+     * performed. If this can't be done, the method falls back to the default
+     * <code>Toolkit.getDefaultToolkit().getScreenSize()</code>, although such
+     * method doesn't exclude the taskbar area.
+     * @return A Rectangle with the usable area for maximized windows.
+     * @author DragShot
+     */
+    public static Rectangle getScreenSize(){
+        Rectangle screen;
+        if (useSun2D) { //Use sun.java2d.SunGraphicsEnvironment.getUsableBounds()
+            try {
+                Frame frame = new Frame();
+                GraphicsConfiguration config = frame.getGraphicsConfiguration();
+                screen = (Rectangle)getUsableBounds.invoke(null, config.getDevice());
+                frame.dispose();
+            } catch (Exception ex) {
+                //If something doesn't work, fallback to Toolkit
+                Dimension size = Toolkit.getDefaultToolkit().getScreenSize();
+                screen = new Rectangle(0, 0, size.width, size.height);
+            }
+        } else { //Do it the traditional way
+            Dimension size = Toolkit.getDefaultToolkit().getScreenSize();
+            screen = new Rectangle(0, 0, size.width, size.height);
+        }
+        return screen;
+    }
+    
+    /**
+     * Checks if the translucency effect is supported. Java 6 does not support
+     * this. Only Java 7 and higher VMs might do, depending of the Graphics
+     * Environment and OS.
+     * @return <code>true</code> if translucency is supported,
+     *         <code>false</code> otherwise.
+     * @author DragShot
+     */
+    public static boolean isTranslucencySupported(){
+        boolean nativeTrans;
+        if (System.getProperty("java.version").contains("1.6")) {
+            System.err.println("Per-pixel translucency is currently not "
```

Comment:
They don't? Wow that's weird. I wouldn't expect that from Linux.

Anyway, you said the minimum source-level is 1.7 right? Then I can safely remove it.

I did some tests in a spare Windows2K VM I had at work, thanks to it I figured out some issues there were with font antialiasing when it is disabled by default by your Desktop configuration. As it is a quite an old OS, I needed to compile to Java 6 in order to run my tests, and `GraphicsDevice.isWindowTranslucencySupported()` doesn't even exist in Java 6.

Besides, the code actually comes from [DS Desktop Notify](https://github.com/DragShot/DS-Desktop-Notify).